### PR TITLE
Support passthrough ops for batch norm fusion (#4776)

### PR DESCRIPTION
**Pull request by DrJessop**
Summary:

Prior to this diff, the fuse batch norm with conv function in prepare only supported conv immediately followed by batch norm. Now, we also support fusing through a non-channel slice. We may consider adding more passthrough ops when needed.

Differential Revision: D116357358

### CODE REVIEW

**Comment by vkuzo** [comment]
@claude review

**Comment by vkuzo** [comment]
lgtm. Could you make sure OSS CI is green, especially the `ruff` piece

**Comment by DrJessop** [comment]
https://github.com/pytorch/ao/actions/runs/32752092932/job/97511173340?pr=4776 Keeps failing with "Node 20 is being deprecated", which is unrelated to my diff. Any thoughts on how to move forward to get a green CI?

**Review by vkuzo — APPROVED** [review]
CI errors look unrelated, lgtm, thank you!
